src/lib/dialog/dialog.ts

```diff
@@ -52,6 +48,34 @@ export class MdDialog {
       @Optional() @SkipSelf() private _parentDialog: MdDialog) { }
 
   /**
+   * Opens a modal dialog containing the given TemplateRef.
+   * @param TemplateRef to load into the dialog
+   * @param config Extra configuration options.
+   * @returns Reference to the newly-opened dialog.
+   */
+  openFromTemplateRef<T>(templateRef: TemplateRef<T>, config?: MdDialogConfig): MdDialogRef<T> {
+    config = _applyConfigDefaults(config);
+
+    let overlayRef = this._createOverlay(config);
```

Comment:
Similar problem as with https://github.com/angular/material2/pull/2853#discussion_r98502685 I don't think I can do that unless I have ViewContainerRef, which I don't unless I require the user to pass me `TemplatePortal`..